src/core/inlines.js

```diff
@@ -86,6 +86,21 @@ function inlineAbbrMatches(matched, txt, abbrMap) {
     : hyperHTML`<abbr title="${abbrMap.get(matched)}">${matched}</abbr>`;
 }
 
+/**
+ * @example |Interface:varName| => <var data-type="Interface">varName</var>
+ * @example |varName| => <var>varName</var>
+ * @param {string} matched
+ */
+function inlineVariableMatches(matched) {
+  // remove "|" at the beginning and at the end, then split at an optional `:`
+  const matches = matched.slice(1, -1).split(":", 2);
+  if (matches.length === 2) {
+    const [interfaceName, varName] = matches;
+    return hyperHTML`<var data-type="${interfaceName}">${varName}</var>`;
```

Comment:
AFAIK currently `data-type` is used to represent the type of IDL definitions, so this looks somewhat confusing. Maybe `data-parent`, `data-scope`, or `data-context`?

Reply:
Oh wait, it's not really a scope?

Reply:
Yeah, we might have a conflict. I need to check how we use data-type in idl.

Reply:
I guess it depends on you definition of scope.

Reply:
I thought I misunderstood and it literally represented the variable's type. And now I'm confused 😵

Reply:
Just to be clear, the use case is:

`Let |DOMString:name| be “hello”.`

Reply:
Okay then, `data-type` looks good as-is. But then the syntax looks too much like `DOMString.name` for me...

Reply:
Sorry for the confusion. I realize now we didn’t give enough context/explanation of what we were trying to achieve. I’ll add a proper bug description tomorrow so it’s more clear.

Reply:
> looks too much like DOMString.name

We were also thinking of using c++ style `::` namespaces instead of just `:`. I don’t feel too strongly about it.

Reply:
At first sight I thought it's something like C++ namespaces 🤣

Reply:
Heh, I actually took inspiration from c++... as all DOM interfaces end up being c++ anyway.

Reply:
>We were also thinking of using c++ style :: namespaces

Based on what I read, this is not about scoping something but about typing a variable, right? Maybe TypeScript-like `|name: DOMString|` would be good enough then.

Reply:
> typing a variable, right? 

Correct. 

> |name: DOMString|

I guess because I’ve not used too much typescript, I still find the above odd to look at... I kinda want the type first, then the variable name. Maybe we don’t need the “:” at all... we can just use a space:

Let |DOMString name| be “hello”.

Reply:
I think I've seen some variables that have spaces in their name.

Reply:
Blasphemy! But yes, I’ve seen it too. Could be an opportunity to discourage that... or in that case, the typescript approach could work rather well.

Reply:
Note the example above is all sorts of confusing... is it a concept used as a variable? That should really be an internal slot.

Reply:
Ok, so in ReSpec’s tradition of not being overly opinionated, we should go with the typescript syntax.